nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -106,7 +106,7 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
             return dataset
 
         shuffle = config['shuffle']
-        if config.get('is_tarred', False):
+        if isinstance(dataset, torch.utils.data.IterableDataset):
```

Comment:
dataset can be None. do complete checks

Reply:
I don't think dataset can be None. If you look 5 lines above this one, there is: